crypto/rand/drbg_lib.c

```diff
@@ -164,7 +229,14 @@ int RAND_DRBG_instantiate(RAND_DRBG *drbg,
     }
 
     drbg->state = DRBG_READY;
-    drbg->reseed_counter = 1;
+    drbg->generate_counter = 0;
+    drbg->reseed_time = time(NULL);
+    if (drbg->reseed_counter > 0) {
```

Comment:
Why this if?

Reply:
I can not find any place where this gets set value different than 0.

Reply:
You are completely right, good catch! This got lost somehow  in my last overhaul when I untangled time-based reseeding and seed propagation from master to public and private DRBG. The `drbg->reseed_time` used to be non-zero, but the `drbg->reseed_counter` isn't anymore. It needs to be set explicitely to 1 in `drbg_setup()` in order to enable seed propagation for the shared DRBGs.

What worries me a bit that this glitch was not noticed by my tests which explicitely check whether seed propagation works. I'll have to look into this.

Reply:
The problem in the test is of course that previous tests set those counters to something other than 0.

Reply:
Yes, I noticed. Meanwhile the test has been modified such that the reseed_counters are not modified anymore before the initial call to test_drbg_reseed() (see drbgtest.c:693).